test/gpu/native/studies/chop/functions.bash

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+
+# Helper functions for logging and marking test start/stop. Formats output in
+# the way the testing infrastructure expects.
```

Comment:
I'm wondering if there's something we can do to avoid copying this for each new external test config we want to add. It was originally added for arkodua, then copied for champs (and maybe modified), and I suspect this is copied from the champs one. It'd be great it we could just have one copy that all external projects use (or more ideally it'd be great if we could reuse parts of our main test infra instead of duplicating that even, but that's more work.)

I'm also wondering if we actually need a custom sub_test for CHOP. I added Arkouda testing with a custom sub_test because it's a very different type of application with it's own testing system and a client/server model. I didn't love adding a custom sub_test (noted in https://github.com/chapel-lang/chapel/pull/14826 and https://github.com/Cray/chapel-private/issues/497) but it did feel like the best way at the time given the differences and the desire to have standalone testing infra in arkouda.

For CHOP though, it seems much more like just a normal chapel program so I'm wondering if we can treat it more like a normal test in the testing infra. My thought would be to have a PRECOMP or maybe an executable SKIPIF that clones the repo and then you have some support files like execopts/graph files that live in a directory above that you then maybe symlink in or overlay in some way. That way I think you wouldn't need a custom sub_test and I _think_ you could just run it as part of normal gpu perf testing instead of adding an additional configuration.

Reply:
I think we can symlink to the files/dirs we care about. https://github.com/chapel-lang/chapel/compare/main...ronawho:chapel:chop-testing-proto is a prototype of that. It symlinks `chplGPU` and the support dirs, and there's a skipif that will try to clone and if the clone fails we'll skip testing.

I think something like that should work and then you can add `chplGPU.{gpu-execopts,gpu-keys,graph}`. I like that it avoids a new testing config and avoids a custom sub_test, but there could be some downsides I'm not seeing.